### DISCUSSION TITLE
Problem with BaseItemsProvider

### DISCUSSION THREAD

**Discussion by rubyyang999** [iTwin/itwinjs-core#3349] [Q&A] [answered]
i'm trying to do Customizing the iTwin Viewer - "The Basics", 
however, when I was constrcting my first UI widget, I have problem like this
![image](https://user-images.githubusercontent.com/59313962/158002918-f6f46aae-eb4a-40f0-b196-42f0b4f7bbc8.png)

myfirstuiprovider .tsx is like this
![image](https://user-images.githubusercontent.com/59313962/158002802-a461c7e2-25b3-49c7-aca3-a516db54d6ca.png)

and my app.tsx
            

>           import { MyFirstUiProvider } from "./MyFirstUiProvider";
                      const uiProviders = [new MyFirstUiProvider()];
                        <Viewer
                                    iTwinId={iTwinId}
                                    iModelId={iModelId}
                                    authClient={authClient}
                                    viewCreatorOptions={viewCreatorOptions}
                                    uiProviders={uiProviders}
                            />

Plz help me~ first time to do this TT thx

**Comment by KrishKash** [comment, marked as answer]
Hi @rubyyang999 ,
I tried to copy your code in sandbox [github_issue_3349](https://www.itwinjs.org/sandboxes/Krishna%20Kashyap/github_issue_3349) and it worked. Please delete your node modules and try again. Or create a sandbox with your issue.

**Reply by rubyyang999** [reply]
Thank you very much!!!!!!!!!!!!!!
I create a sandbox at last!
Because the version changed, there exists lots of problem for the tutorial.
